jax/_src/numpy/ufuncs.py

```diff
@@ -2559,6 +2567,10 @@ def _float_divmod(x1: ArrayLike, x2: ArrayLike) -> tuple[Array, Array]:
   mod = lax.select(ind, mod + x2, mod)
   div = lax.select(ind, div - _constant_like(div, 1), div)
 
+  div = lax.select(_broadcast_to(x2 == 0, div.shape),
+                   lax.full_like(div, fill_value=np.inf) * lax.sign(x1),
+                   div)
+
```

Comment:
We need to double check this logic for when `x2` is `-0.0` (the sign of the returned infinity may be wrong), or when `x1` and `x2` are both zero (when the result should be NaN).

Reply:
Thanks! Modified the `_float_divmod` logic to consider all the possible scenarios to match with NumPy.